tests/crypto_policy_test.go

```diff
@@ -0,0 +1,238 @@
+					OpenSSLCipherNames: []string{},
+				},
+			},
+			disallowedConfigs: []clientTLSOptions{
+				{
+					MaxTLSVersion:      string("1.2"),
+					OpenSSLCipherNames: []string{tls10AllowedCipher},
+				},
+			},
+		},
+		"Modern": {
+			openshiftTLSPolicy: &modern,
+			allowedConfigs: []clientTLSOptions{
+				{
+					MaxTLSVersion:      string("1.3"),
+					OpenSSLCipherNames: []string{},
+				},
+			},
+			disallowedConfigs: []clientTLSOptions{
+				{
+					MaxTLSVersion:      string("1.2"),
+					OpenSSLCipherNames: []string{},
+				},
+			},
+		},
+	}
+)
+
+func getPemCertificate() []byte {
+	if pemCertificate != nil {
+		return pemCertificate
+	}
+	var apiService apiregv1.APIService
+	err := apiClient.Get(ctx, client.ObjectKey{Name: "v1.build.openshift.io"}, &apiService)
+	if err != nil {
+		panic(err)
+	}
+	pemCertificate = apiService.Spec.CABundle
+	return pemCertificate
+}
+
+func tryToAccessEndpoint(pod core.Pod, port, subpath string, tlsConfig clientTLSOptions) error {
+	portInt, err := strconv.ParseUint(port, 10, 16)
+	Expect(err).ToNot(HaveOccurred())
+	conn, err := portForwarder.Connect(&pod, uint16(portInt))
+	Expect(err).ToNot(HaveOccurred())
+	defer conn.Close()
+	tlsVersion, err := tlsConfig.MaxTLSVersionId()
+	if err != nil {
+		return err
+	}
+
+	certPool := x509.NewCertPool()
+
+	pemCert := getPemCertificate()
+	certPool.AppendCertsFromPEM(pemCert)
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
+				return conn, nil
+			},
+			TLSClientConfig: &tls.Config{CipherSuites: tlsConfig.CipherIDs(nil), MaxVersion: tlsVersion, RootCAs: certPool},
+		},
+	}
+
+	if subpath != "" {
+		subpath = "/" + subpath
+	}
+	_, err = client.Get(fmt.Sprintf("https://ssp-webhook-service.kubevirt.svc:%s%s", port, subpath))
+	return err
+}
+
+func testTLSEndpointAccessible(pod core.Pod, port, subpath string, tlsConfig clientTLSOptions) {
+	err := tryToAccessEndpoint(pod, port, subpath, tlsConfig)
+	Expect(err).ToNot(HaveOccurred(), "Can't access pod %s, at port %s, with tlsConfig %#v", pod.Name, port, tlsConfig)
+}
+
+func testTLSEndpointNotAccessible(pod core.Pod, port, subpath string, tlsConfig clientTLSOptions) {
+	err := tryToAccessEndpoint(pod, port, subpath, tlsConfig)
+	version, _ := tlsConfig.MaxTLSVersionId()
+	Expect(err).To(HaveOccurred(), "Should not have been able to access pod %s, at port %s, with tlsConfig %#v, %#v, %#v", pod.Name, port, tlsConfig, version, tlsConfig.CipherIDs(nil))
+}
+
+func testEndpointAccessabilityWithTLS(pod core.Pod, port, subpath string, allowedConfigs, disallowedConfigs []clientTLSOptions) {
+	for _, config := range allowedConfigs {
+		testTLSEndpointAccessible(pod, port, subpath, config)
+	}
+	for _, config := range disallowedConfigs {
+		testTLSEndpointNotAccessible(pod, port, subpath, config)
+	}
+}
+
+func testMetricsEndpoint(pod core.Pod, allowedConfigs, disallowedConfigs []clientTLSOptions) {
```

Comment:
I would favor inlining this into `testSettingTLSConfig`.

Reply:
I prefer not to, for adding future tests it is easier to have those separate.